### PR TITLE
Fix #772: Lowfi Landscape: AppLanguageActivity #772

### LINKED ISSUE #772: Lowfi Landscape: AppLanguageActivity
Mock: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/2b8387b0-5ec6-4e42-bffa-c039128b2f12/OP-App-Language-/

`AppLanguageActivity` should support landscape mode. Also add test cases for landscape mode too.

This issue is currently blocked because the landscape mode mocks are not ready yet.

### PR DESCRIPTION
**Pull request by MohamedMedhat1998**
## Explanation

-Fixes #772: Lowfi Landscape: AppLanguageActivity   

## Before  
The `AppLanguageActivity` was locked to the portrait mode  
Screenshot (129)
## After
The `AppLanguageActivity` now supports the landscape mode  
Screenshot (128)
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.
- [x] Implementing a landscape for `AppLanguageActivity`
- [x] Unit tests for `AppLanguageActivity`

### CODE REVIEW

**Comment by MohamedMedhat1998** [comment]
@rt4914 please review this PR

File: app/src/main/java/org/oppia/app/options/AppLanguageActivity.kt [hunk 1]
```diff
@@ -44,4 +50,9 @@ class AppLanguageActivity : InjectableAppCompatActivity() {
     setResult(REQUEST_CODE_APP_LANGUAGE, intent)
     finish()
   }
+
+  override fun onSaveInstanceState(outState: Bundle) {
+    super.onSaveInstanceState(outState)
+    outState.putSerializable(KEY_LANGUAGE_SELECTION_ADAPTER ,appLanguageActivityPresenter.getLanguageSelectionAdapter())
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/options/AppLanguageActivity.kt]: Spacing is incorrect there should be space after `,` and not before that:
`outState.putSerializable(KEY_LANGUAGE_SELECTION_ADAPTER, appLanguageActivityPresenter.getLanguageSelectionAdapter())`

File: app/src/main/java/org/oppia/app/options/AppLanguageActivityPresenter.kt [hunk 2]
```diff
@@ -47,4 +47,8 @@ class AppLanguageActivityPresenter @Inject constructor(private val activity: App
     languageSelectionAdapter.setDefaultLanguageSelected(prefSummaryValue = prefSummaryValue)
   }
 
+  fun getLanguageSelectionAdapter(): LanguageSelectionAdapter{
+    return languageSelectionAdapter
+  }
+
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/options/AppLanguageActivityPresenter.kt]: Remove this extra blank line

File: app/src/main/java/org/oppia/app/options/AppLanguageActivityPresenter.kt [hunk 3]
```diff
@@ -14,10 +14,10 @@ class AppLanguageActivityPresenter @Inject constructor(private val activity: App
   private lateinit var languageSelectionAdapter: LanguageSelectionAdapter
   private lateinit var prefSummaryValue: String
 
-  fun handleOnCreate(prefKey: String, prefSummaryValue: String) {
+  fun handleOnCreate(prefKey: String, prefSummaryValue: String,savedAdapter : LanguageSelectionAdapter? = null) {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/options/AppLanguageActivityPresenter.kt]: Spacing issue:
`fun handleOnCreate(prefKey: String, prefSummaryValue: String, savedAdapter: LanguageSelectionAdapter? = null) {`

File: app/src/main/java/org/oppia/app/options/AppLanguageActivityPresenter.kt [hunk 4]
```diff
@@ -47,4 +47,8 @@ class AppLanguageActivityPresenter @Inject constructor(private val activity: App
     languageSelectionAdapter.setDefaultLanguageSelected(prefSummaryValue = prefSummaryValue)
   }
 
+  fun getLanguageSelectionAdapter(): LanguageSelectionAdapter{
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/options/AppLanguageActivityPresenter.kt]: Spacing issue:
`fun getLanguageSelected(): String {`

File: app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt [hunk 5]
```diff
@@ -246,6 +267,30 @@ class OptionsFragmentTest {
...
 
+  @Test
+  fun testOptionFragment_clickAppLanguage_configurationChange_changeAppLanguageHindiSuccessfully() {
+    launch<OptionsActivity>(createOptionActivityIntent(0)).use {
+      onView(atPositionOnView(R.id.options_recyclerview, 1, R.id.app_language_item_layout)).perform(
+        click()
+      )
+      onView(withId(R.id.language_recycler_view))
+        .perform(
+          RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt]: Nit:
`actionOnItemAtPosition<RecyclerView.ViewHolder>(`

File: app/src/main/java/org/oppia/app/options/LanguageSelectionAdapter.kt [hunk 6]
```diff
@@ -7,12 +7,13 @@ import androidx.databinding.library.baseAdapters.BR
...
 import org.oppia.app.R
 import org.oppia.app.databinding.LanguageItemsBinding
+import java.io.Serializable
 
 // TODO(#216): Make use of generic data-binding-enabled RecyclerView adapter.
 
 /** Adapter to bind languages to [RecyclerView] inside [AppLanguageActivityPresenter] and [DefaultAudioActivityPresenter]. */
 class LanguageSelectionAdapter(private val prefKey: String) :
-  RecyclerView.Adapter<LanguageSelectionAdapter.LanguageViewHolder>() {
+  RecyclerView.Adapter<LanguageSelectionAdapter.LanguageViewHolder>(),Serializable {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/options/LanguageSelectionAdapter.kt]: We don't want the entire Adapter to be Serializable. Instead check the implementation of `onSavedInstance` in `DefaultAudioActivity`.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Suggested approach changes.

File: app/src/main/java/org/oppia/app/options/AppLanguageActivity.kt [hunk 7]
```diff
@@ -19,7 +20,11 @@ class AppLanguageActivity : InjectableAppCompatActivity() {
     super.onCreate(savedInstanceState)
     activityComponent.inject(this)
     prefKey = intent.getStringExtra(KEY_APP_LANGUAGE_PREFERENCE_TITLE)
-    prefSummaryValue = intent.getStringExtra(KEY_APP_LANGUAGE_PREFERENCE_SUMMARY_VALUE)
+    prefSummaryValue = if(savedInstanceState == null){
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/options/AppLanguageActivity.kt]: Spacing issue:
```
    prefSummaryValue = if (savedInstanceState == null) {
      intent.getStringExtra(KEY_APP_LANGUAGE_PREFERENCE_SUMMARY_VALUE)
    } else {
      savedInstanceState.get(KEY_SELECTED_LANGUAGE) as String
    }
    appLanguageActivityPresenter.handleOnCreate(prefKey, prefSummaryValue)
```

**Review by rt4914 — CHANGES_REQUESTED** [review]
Everything looks great. Just one nit spacing issue change suggested.

**Review by rt4914 — APPROVED** [review]
@MohamedMedhat1998 Everything is perfect in this PR but if you see at the bottom of the PR `non-flaky_tests` are failing and because of we cannot merge this PR. So to fix that we can do following steps:
Merge your branch with latest code from upstream/develop so that your current branch is ahead of `oppia-android/develop` and push that code. Once that is done mostly the CircleCI tests will pass. So that we can merge your PR. Thanks.

**Comment by rt4914** [comment]
Closing this PR as it has been replicated on #859

**Pull request closed without merging**